File: src/main/java/com/shop/domain/item/dto/ItemResponse.java
package com.shop.domain.item.dto;

import java.util.List;

import com.shop.domain.category.dto.CategoryResponse;

import lombok.Getter;
import lombok.Setter;

@Setter
@Getter
public class ItemResponse {
    // Item
    private Long itemId;
    private Long price;
    private String createdAt;
    private String modifiedAt;

    // Mart
    private Long martId;
    private String martName;

    // Product
    private Long productId;
    private String productName;
    private String code;

    // Category
    private List<CategoryResponse> categories;

    // Brand
    private Long brandId;
    private String brandName;
}

File: src/utils/api/item/response/ItemResponse.ts
import type { Dayjs } from "dayjs";
import type { CategoryResponse } from "../../category/response/CategoryResponse";

export type ItemResponse = {
  itemId: number;
  price: number;
  createdAt: Dayjs;
  modifiedAt: Dayjs;
  martId: number;
  martName: string;
  productId: number;
  productName: string;
  barcode: string;
  categories: CategoryResponse[];
};
